box64-bundle-x86-libs.sh

```diff
@@ -1,121 +1,123 @@
+            ${unzip} -o "${dir_tmp}"/bundle-pkgs/"${file}"
+            tar -xf install.tar.xz
+            ;;
+        rpm)
+            echo "I: RPM (Fedora) detected (${file})"
+            ${rpm2cpio} "${dir_tmp}"/bundle-pkgs/"${file}" | cpio -idmv
+            ;;
+        xbps)
+            echo "I: XBPS (Void Linux) package detected (${file})"
+            tar -xf "${dir_tmp}"/bundle-pkgs/"${file}"
+            ;;
+    esac
 done
 
-# Remove libraries that cannot be emulated.
+# reorganize the files
+cd "${dir_tmp}"/bundle-libs
+mkdir -p "${dir_tmp}"/bundle-libs/box64-i386-linux-gnu
+mv lib/*.so* usr/lib/*.so* usr/lib/i386-linux-gnu/*.so* usr/lib32/*.so* \
+    "${dir_tmp}"/bundle-libs/box64-i386-linux-gnu
+mkdir -p "${dir_tmp}"/bundle-libs/box64-x86_64-linux-gnu
+mv lib64/*.so* usr/lib/x86_64-linux-gnu/*.so* usr/lib64/*.so* \
+    "${dir_tmp}"/bundle-libs/box64-x86_64-linux-gnu
+rm -f ./*.tar.* debian-binary files.xml metadata.xml
+rm -rf ./lib* etc run sbin usr var
+mkdir -p usr/lib
+mv box64-*-linux-gnu usr/lib
+
+# remove libraries that cannot be emulated
 # https://github.com/ptitSeb/box64/blob/v0.3.6/src/librarian/library.c#L433
-for lib in libc.so libpthread.so librt.so libGL.so libGL.so libX11.so \
-  libasound.so libdl.so libm.so libbsd.so libutil.so libresolv.so \
-  libXxf86vm.so libXinerama.so libXrandr.so libXext.so libXfixes.so libXcursor.so \
-  libXrender.so  libXft.so libXi.so libXss.so libXpm.so libXau.so libXdmcp.so \
-  libX11-xcb.so libxcb.so libxcb-xfixes.so libxcb-shape.so libxcb-shm.so libxcb-randr.so \
-  libxcb-image.so libxcb-keysyms.so libxcb-xtest.so libxcb-glx.so libxcb-dri2.so libxcb-dri3.so \
-  libXtst.so libXt.so libXcomposite.so libXdamage.so libXmu.so libxkbcommon.so \
-  libxkbcommon-x11.so libpulse-simple.so libpulse.so libvulkan.so libvulkan.so \
-  ld-linux-x86-64.so crashhandler.so libtcmalloc_minimal.so libtcmalloc_minimal.so libanl.so \
-  ld-linux.so ld-linux.so libthread_db.so
-    do find "${dir_tmp_local}" -name "${lib}"* -delete
+set -- libc libpthread librt libGL libX11 libasound libdl libm libbsd libutil \
+  libresolv libXxf86vm libXinerama libXrandr libXext libXfixes libXcursor \
+  libXrender libXft libXi libXss libXpm libXau libXdmcp libX11-xcb libxcb \
+  libxcb-xfixes libxcb-shape libxcb-shm libxcb-randr libxcb-image \
+  libxcb-keysyms libxcb-xtest libxcb-glx libxcb-dri2 libxcb-dri3 libXtst libXt \
+  libXcomposite libXdamage libXmu libxkbcommon libxkbcommon-x11 \
+  libpulse-simple libpulse libvulkan ld-linux-x86-64 crashhandler \
+  libtcmalloc_minimal libanl ld-linux libthread_db
+for file in "$@"; do
+    rm -f "${dir_tmp}"/bundle-libs/usr/lib/box64-*-linux-gnu/"${file}".so*
 done
 
-if find "${dir_tmp_local}"/bundle-libs/ -type l ! -exec test -e {} \; -print | grep bundle-libs
-    then echo "Broken symlinks found."
-    cd -
-    echo rm -r -f "${dir_tmp_local}"
+# check broken symlinks
+if [ -n "$(find "${dir_tmp}"/bundle-libs -type l ! -exec test -e {} \; -print)" ]; then
```

Comment:
The problem with this is that the `$()` hides the output so it's not clear what symlinks are broken.

Reply:
I had the same in mind. We can call it twice:
```
if [ -n "$(find "${dir_tmp}"/bundle-libs -type l ! -exec test -e {} \; -print)" ]; then
    echo "E: Broken symlinks found"
    # display the output
    find "${dir_tmp}"/bundle-libs -type l ! -exec test -e {} \; -print
    exit 1
fi
```